filebeat/module/nginx/ingress_controller/ingest/pipeline.yml

```diff
@@ -139,18 +140,36 @@ processors:
           for (def item : ctx.nginx.ingress_controller.upstream_address_list) {
             last_upstream = item;
           }
-          StringTokenizer tok = new StringTokenizer(last_upstream, ":");
-          if (tok.countTokens()>1) {
-            ctx.nginx.ingress_controller.upstream.ip = tok.nextToken();
-            ctx.nginx.ingress_controller.upstream.port = Integer.parseInt(tok.nextToken());
-          } else {
-            ctx.nginx.ingress_controller.upstream.ip = last_upstream;
-          }
+
+          ctx.nginx.ingress_controller.upstream.address = last_upstream;
         }
         catch (Exception e) {
-          ctx.nginx.ingress_controller.upstream.ip = null;
-          ctx.nginx.ingress_controller.upstream.port = null;
+          ctx.nginx.ingress_controller.upstream.address = null;
         }
+  - grok:
+      field: nginx.ingress_controller.upstream.address
+      patterns:
+        - "^%{IPV4:nginx.ingress_controller.upstream.ip}:%{NUMBER:nginx.ingress_controller.upstream.port}$"
+        - "^\\[%{IPV6:nginx.ingress_controller.upstream.ip}\\]:%{NUMBER:nginx.ingress_controller.upstream.port}$"
+        - "^%{IPV6NOCOMPRESS:nginx.ingress_controller.upstream.ip}:%{NUMBER:nginx.ingress_controller.upstream.port}$"
+        - "^%{IPV6:nginx.ingress_controller.upstream.ip}%{IPV6PORTSEP}%{NUMBER:nginx.ingress_controller.upstream.port}$"
+        - "^%{IPV6:nginx.ingress_controller.upstream.ip}%{IPV6PORTSEP}%{POSINT:nginx.ingress_controller.upstream.port}$"
+      pattern_definitions:
+        IPV6NOCOMPRESS: '([0-9A-Fa-f]{1,4}:){7}[0-9A-Fa-f]{1,4}'
+        IPV6PORTSEP: '(?: port |[p#.])'
+      ignore_missing: true
+      ignore_failure: true
+  - convert:
+      field: nginx.ingress_controller.upstream.ip
```

Comment:
If the `convert` fails then you should handle it by removing the invalid IP from this field via `on_failure`. If you leave the bad value in a field mapped as an IP then the event will fail to index unless the mapping for the field includes `ignore_malformed: true`.

Same goes for the upstream.port.

Reply:
I fixed the pipeline now and added some tests